spec/services/tax_rate_finder_spec.rb

```diff
@@ -0,0 +1,73 @@
+require 'spec_helper'
+
+describe TaxRateFinder do
+  describe "getting the corresponding tax rate" do
+    let(:amount) { BigDecimal(100) }
+    let(:included_tax) { BigDecimal(20) }
+    let(:tax_rate) { create_rate(0.2) }
+    let(:close_tax_rate) { create_rate(0.25) }
+    let(:other_tax_rate) { create_rate(0.3) }
+    let(:tax_category) { create(:tax_category, tax_rates: [tax_rate]) }
+    # This zone is used by :order_with_taxes and needs to match it
+    let(:zone) { create(:zone, name: "GlobalZone") }
+    let(:shipment) { create(:shipment) }
+    let(:enterprise_fee) { create(:enterprise_fee, tax_category: tax_category) }
+    let(:order) { create(:order_with_taxes) }
+
+    it "finds the tax rate of a shipping fee" do
+      rates = TaxRateFinder.new.tax_rates(
+        tax_rate,
+        shipment,
+        amount,
+        included_tax
+      )
+      expect(rates).to eq [tax_rate]
+    end
+
+    it "finds a close match" do
+      close_tax_rate
```

Comment:
If this tax rate is only used here why not having the following?

```suggestion
      close_tax_rate = create_rate(0.25)
```

It's weird for the reader to see a method call named after a noun without an assignment. Also, having a let at the top it makes you think it's an object that will be used throughout all tests, while it's not.

Reply:
Thank you for the feedback. I followed your suggestion and discovered that the spec was testing the wrong thing. Have a look at the last commit and let us know if you are happy with the pull request.